lib/Driver/ToolChains.cpp

```diff
@@ -261,7 +261,11 @@ static void addCommonFrontendArgs(const ToolChain &TC, const OutputInfo &OI,
   // -g implies -enable-anonymous-context-mangled-names, because the extra
   // metadata aids debugging.
   if (inputArgs.hasArg(options::OPT_g)) {
-    arguments.push_back("-enable-anonymous-context-mangled-names");
+    // But don't add the option in optimized builds: it would prevent dead code
+    // stripping of unused metadata.
+    auto OptArg = inputArgs.getLastArgNoClaim(options::OPT_O_Group);
+    if (!OptArg || OptArg->getOption().getID() == options::OPT_Onone)
```

Comment:
Nitpick: `OptArg->getOption().matches(options::OPT_Onone)`. That'll handle aliases for -Onone if we ever add them.

Reply:
Thanks, I changed it.